python_modules/libraries/dagster-dbt/dbt_packages/dagster/macros/log_columns_in_relation.sql

```diff
@@ -1,14 +1,31 @@
 {% macro log_columns_in_relation() %}
```

Comment:
Is my understanding correct that this logs the columns in the produced table as well as all of the columns of all of its parent tables? If so:
- I think it's worth adding some comments to document that that's what it's doing, along with an example of the format that they're logged in
- I think I would expect the parents to be nested in some way, to distinguish them from the table itself
- It seems like the name "log_columns_in_relation" isn't entirely accurate? Because we're actually logging columns in multiple relations?
- What if someone wants to collect column schema but not lineage (e.g. for perf reasons)?

Reply:
I renamed the macro to `log_column_level_metadata` and added more extensive documentation about what's happening. This includes the format of the resulting structured log that's emitted from the macro.

> What if someone wants to collect column schema but not lineage (e.g. for perf reasons)?

This PR is already pretty large, so I'm addressing this in another change.

Reply:
Here's the change to allow the user to disable collection of parent schema metadata (and thus column dependencies): https://github.com/dagster-io/dagster/pull/20548